spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -772,13 +773,20 @@ def change_clients_pdb_use_exclamation_mark(self, value):
                 value)
 
     @on_conf_change(option=[
+        'symbolic_math', 'hide_cmd_windows',
         'startup/run_lines', 'startup/use_run_file', 'startup/run_file',
-        'pylab', 'pylab/backend', 'symbolic_math', 'hide_cmd_windows'])
+        'pylab', 'pylab/backend'])
     def change_possible_restart_conf(self, option, value):
         """Apply options that possibly require a kernel restart."""
-        if not self.get_current_client():
+        # Check that we are not triggering validations in the initial
+        # notification sent when Spyder is starting or when another option
+        # already required a restart and the restart dialog was shown
+        if option in self.initial_conf_options:
+            self.initial_conf_options.remove(option)
             return
 
+        # Also check if an actual client is available
+
```

Comment:
I don't understand this comment. Could relocate it to a better place? Or, if that's not possible, could you remove it?